### ISSUE TITLE
FaultOutputWriter: RECPOINT not associated

### ISSUE THREAD

**Issue by sebwolf-de** [SeisSol/SeisSol#150]
**Describe the bug**
For larger number (depends on the size of the mesh, in my exampe for n>=4) of nodes SeisSol stops when initializing the Fault Output with a FORTRAN error message. Instead of terminating it hangs.
When I use a low number of nodes everything works fine. I guess the problem is related to MPI ranks which do not contain parts of the fault.

**Expected behavior**
SeisSol should initialize the Fault Output correctly and start the calculations.

**To Reproduce**
1. Which version do you use? Provide branch and commit id.
Branch master, commit 04a70cf562818e23cfd2f8a983714ad3f1e946d0
2. Which build settings do you use? Which compiler version do you use? 
```
  compileMode                 = 'debug'
  parallelization             = 'hybrid'
  generatedKernels            = 'yes'
  order = 4
  equations='elastic'
  netcdf='yes'
  hdf5='yes'
  metis='yes'
  netcdfDir=os.environ['NETCDF_BASE']
  hdf5Dir=os.environ['HDF5_BASE']
  asagi='yes'
  zlibDir='/hppfs/work/pr45fi/di73yeq4/myLibs/ASAGI/build'
  arch = 'sskx'
  commThread ='yes'
  logLevel                    = 'debug'
  logLevel0                   = 'debug'
```
Compiler is intel 2019.
3. On which machine does your problem occur? If on a cluster: Which modules are loaded?
supermuc-NG
Loaded modules: 
```
admin/1.0
tempdir/1.0
spack/release/19.1
lrz/default
scons/3.0.1
cmake/3.6
python/3.6_intel
slurm_setup/default
parmetis/4.0.3-intel-impi-i64-r64
metis/5.1.0-intel-i64-r64
netcdf/4.6.1-intel-impi-hdf5v1.8-parallel
libszip/2.1.1
numactl/2.0.11
hdf5/1.8.20-intel-impi-threadsafe
intel/19.0
mkl/2019
mpi.intel/2019
itac/2019
advisor_xe/2019
amplifier_xe/2019
inspector_xe/2019
devEnv/Intel/2019
```

4. Provide parameter/material files.
The problem only occurs if OutputPointType = 4 or 5
[parameters.txt](https://github.com/SeisSol/SeisSol/files/3757124/parameters.txt)

**Screenshots/Console output**
```
Tue Oct 22 18:29:00, Info:  Mapping receivers to LTS cells... 
forrtl: severe (408): fort: (7): Attempt to use pointer RECPOINT when it is not associated with a target

Image              PC                Routine            Line        Source             
SeisSol_debug_gen  00000000011F9C26  Unknown               Unknown  Unknown
SeisSol_debug_gen  0000000000D1D482  ini_faultoutput_m         970  ini_faultoutput.f90
SeisSol_debug_gen  0000000000CA6479  inioutput_seissol         178  inioutput_seissol.f90
SeisSol_debug_gen  0000000000DF914D  fortran_main              207  seissolxx.f90
SeisSol_debug_gen  0000000000E53117  main                       56  main.cpp
libc-2.22.so       00007F17C36AB725  __libc_start_main     Unknown  Unknown
SeisSol_debug_gen  00000000006EC0F9  Unknown               Unknown  Unknown
forrtl: severe (408): fort: (7): Attempt to use pointer RECPOINT when it is not associated with a target
```

**Comment by Thomas-Ulrich** [comment]
could this node dependance be related to a mpi communicator problem?

**Comment by sebwolf-de** [comment]
Lukas and I found a workaround: https://github.com/SeisSol/SeisSol/tree/fix-unassociated-recpoint
It does not really have to do with MPI, but actually with not allocating memory where needed.

**Comment by krenzland** [comment]
Is there actually a reason why we didn't merge this?

**Comment by sebwolf-de** [comment]
I thought we wanted to investigate this further...

**Comment by sebwolf-de** [comment]
Fixed by #185

**Comment by AliceGabriel** [comment]
Great!
